sei-db/sc/memiavl/db_test.go

```diff
@@ -182,7 +183,10 @@ func TestSnapshotTriggerOnIntervalDiff(t *testing.T) {
 		require.EqualValues(t, 0, db.MultiTree.SnapshotVersion())
 	}
 
-	// Height 5 should trigger rewrite
+	// Wait for minimum time interval to elapse (1 second + buffer)
+	time.Sleep(1100 * time.Millisecond)
```

Comment:
I am worried that this test is going to be flaky as a result of this kind of sleep.

Instead write a function that checks the condition you want is met and run it with `require.Eventually`?